src/Compilers/CSharp/Portable/Symbols/Tuples/TupleTypeSymbol.cs

```diff
@@ -261,7 +261,8 @@ private static void ReportUseSiteAndObsoleteDiagnostics(CSharpSyntaxNode? syntax
         /// </summary>
         internal static void VerifyTupleTypePresent(int cardinality, CSharpSyntaxNode? syntax, CSharpCompilation compilation, DiagnosticBag diagnostics)
         {
-            Debug.Assert(diagnostics is object && syntax is object);
+            RoslynDebug.Assert(diagnostics is object);
```

Comment:
>RoslynDebug [](start = 12, length = 11)

Did the assertions not work in the `&&` form?  If so, that is worrying.

Reply:
The assertions work with `&&` as well but it seemed more useful to test the parts separately to narrow down any failure.

---
In reply to: [366012889](https://github.com/dotnet/roslyn/pull/40892#discussion_r366012889) [](ancestors = 366012889)